src/ispypsa/templater/helpers.py

```diff
@@ -4,70 +4,126 @@
+    spellings will cause the matching to break.
 
     Args:
-        name_series: :class:`pandas.Series` with names to be matched with values in
-            `choices`
-        choices: Iterable of `choices` that are replacement values
-        threshold: Threshold to exceed for replacement. Between 0 and 100
-        task_desc: Task description to include in logging information
+        strings_to_match: set of strings to find a match for in the set of choices.
+        choices: set of strings to choose from when finding matches.
         not_match: optional. Defaults to "existing". If "existing", wherever a match
             that exceeds the threshold does not exist, the existing value is retained.
             If any other string, this will be used to replace the existing value
             where a match that exceeds the threshold does not exist.
+        threshold: match quality threshold to exceed for replacement. Between 0 and 100
+
     Returns:
-        :class:`pandas.Series` with selective fuzzy matching
+        dict: dict matching strings to the choice they matched with.
     """
-    if not_match == "existing":
-        matched_series = name_series.apply(
-            lambda x: process.extractOne(x, choices)[0]
-            if process.extractOne(x, choices)[1] > threshold
-            else x
-        )
-    else:
-        matched_series = name_series.apply(
-            lambda x: process.extractOne(x, choices)[0]
-            if process.extractOne(x, choices)[1] > threshold
-            else not_match
-        )
-    _log_fuzzy_match(name_series, matched_series, task_desc)
-    return matched_series
+    # if len(strings_to_match) > len(choices):
+    #     raise ValueError("To many strings to match.")
+
+    matches = []
+
+    # Keep track of which strings and choices haven't formed a match yet.
+    remaining_strings_to_match = strings_to_match
+    remaining_choices = choices
+
+    # Find and remove exact matches
+    exact_matches = remaining_strings_to_match.intersection(remaining_choices)
+    for s in exact_matches:
+        matches.append((s, s))
+        remaining_strings_to_match.remove(s)
+        remaining_choices.remove(s)
+
+    # Convert remaining sets to lists for index access
+    remaining_strings_to_match_list = list(remaining_strings_to_match)
+    remaining_choices_list = list(remaining_choices)
+
+    scores = []
+
+    # For remaining strings, use greedy approach with fuzzy matching
+    while remaining_strings_to_match_list and remaining_choices_list:
+        best_score = -1
+        best_pair = None
+
+        # Find the highest similarity score among remaining pairs
+        for i, str_a in enumerate(remaining_strings_to_match_list):
+            for j, str_b in enumerate(remaining_choices_list):
+                score = fuzz.ratio(str_a, str_b)
+                scores.append(score)
+                if score > best_score and score >= threshold:
```

Comment:
Threshold is actually needed to prevent bad matching where some static property values for some technologies (e.g. hydrogen turbines) do not exist
![image](https://github.com/user-attachments/assets/e00317ac-e20b-478e-9fb2-ac7df19f421b)